master/internal/agent/slot.go

```diff
@@ -109,10 +110,14 @@ func (s *slot) patch(ctx *actor.Context) {
 			add.ContainerID = &s.container.ID
 		}
 		ctx.Tell(s.resourcePool, add)
-	} else if !s.enabled.Enabled() && s.enabled.deviceAdded {
-		s.enabled.deviceAdded = false
+	} else if !s.enabled.Enabled() {
 		remove := sproto.RemoveDevice{DeviceID: s.deviceID(ctx)}
-		ctx.Tell(s.resourcePool, remove)
+
+		if s.enabled.deviceAdded {
```

Comment:
i guess what i mean is, even if we are draining, if the slot actor stops, we've lost all context on the slot forever. we used to kill task containers in the case (i guess sort of emergency cleanup). i think we still should.

Reply:
I unset draining status above on `PostStop` line 79, and so line 121 should kill the container even if there's no device. is this not sufficient?

Reply:
ah, missed that, that should do it.

Reply:
maybe drop a comment?